destination/src/main/scala/quasar/plugin/sqlserver/destination/CSVCreateSink.scala

```diff
@@ -61,20 +61,20 @@ private[destination] object CsvCreateSink {
     val obj = jdbc.resourcePathRef(pathWithSchema).get // FIXME
 
     val objFragment: Fragment = obj.fold(
-      t => Fragment.const0(SQLServerHygiene.hygienicIdent(t).forSql),
+      t => Fragment.const0(SQLServerHygiene.hygienicIdent(t).forSqlName),
       { case (d, t) =>
-        Fragment.const0(SQLServerHygiene.hygienicIdent(d).forSql) ++
-        fr0"." ++
-        Fragment.const0(SQLServerHygiene.hygienicIdent(t).forSql)
+        Fragment.const0(SQLServerHygiene.hygienicIdent(d).forSqlName) ++
+          fr0"." ++
+          Fragment.const0(SQLServerHygiene.hygienicIdent(t).forSqlName)
       })
 
     val unsafeTableName: String = obj.fold(
-      t => t.asString,
-      { case (_, t) => t.asString})
+      t => SQLServerHygiene.hygienicIdent(t).unsafeForSqlName,
+      { case (_, t) => SQLServerHygiene.hygienicIdent(t).unsafeForSqlName})
 
     val unsafeTableSchema: Option[String] = obj.fold(
```

Comment:
We only need `unsafeTableName` and `unsafeTableSchema` because we have to run a `count(*)` query to determine if the table exists, in order to implement a couple of the `WriteMode`s.

Reply:
You might try using [`JdbcDiscovery#tableExists`](https://github.com/precog/quasar-lib-jdbc/blob/v0.28.3/core/src/main/scala/quasar/lib/jdbc/JdbcDiscovery.scala#L56) instead.

Reply:
ooooo!